### PR TITLE
Tessore randoms

### PR DESCRIPTION
**Pull request by elvinpoole**
Added option to generate randoms using ntessore/healpix package (https://github.com/ntessore/healpix)

Faster than generating per pixel from quadrilateral (~0.3min vs 10min on one core) and will maintain matching to healpix pixel ids when using healpy tools

### CODE REVIEW

**Comment by joezuntz** [comment]
Looks like the tests pass now - is it ready for a review?

**Comment by elvinpoole** [comment]
yes I think so

**Comment by joezuntz** [comment]
Great - @empEvil volunteered at the hack time, so I've assigned him, thanks Eske!

File: txpipe/random_cats.py [hunk 1]
```diff
@@ -168,7 +175,7 @@ def run(self):
             # Generate the random points in each pixel
             ndone = 0
 
-            nvertex = len(vertices)
+            nvertex = npix #number of verticies is the same as number of pixels
```
Comment (empEvil) [inline txpipe/random_cats.py:178]: Just trying to make sure I understand, is this change significant?

**Review by empEvil — COMMENTED** [review]
Great idea, I have a few questions though. Does this break any of the standard examples? If so they should be fixed first! I would suggest that if it isn't too much work this gets made into a sub-class instead, and people can then call the subclass rather than the method to run it.

File: txpipe/random_cats.py (hunk 1 above)
Comment (elvinpoole) [reply txpipe/random_cats.py:178]: no, it changes nothing 
len(vertices) == npix
We can switch it back if that's preferred

**Comment by elvinpoole** [comment]
On the standard examples. I believe they all run as they did before. The default behaviour is currently still to use the old method

**Comment by elvinpoole** [comment]
happy to add it as a sub-class if that makes more sense. I didn't originally as I didn't want us to end up with too many sub-classes for all the different ways of making randoms that might eventually be needed. e.g. depth varying vs uniform, generated from systematic weights, etc

**Comment by empEvil** [comment]
I think this is something we should also get @joezuntz input on

File: txpipe/random_cats.py [hunk 2]
```diff
@@ -231,7 +279,7 @@ def run(self):
                 )
 
                 # Save output to the generic non-binned output
-                index = bin_starts[j] + pix_starts[j, i]
+                index = bin_starts[j] + pix_starts[j, i] #this line was in the quadrilateral method but didn't appear to be doing anything...
```
Comment (joezuntz) [inline txpipe/random_cats.py:282]: Might be left over from an earlier version Feel free to remove!

**Comment by joezuntz** [comment]
I'm happy to leave it as one class for now since there's only two methods here, but lets see how things go later as we develop. Maybe later it will make more sense to split.

**Comment by joezuntz** [comment]
You could add a run of this pipeline to the CI. It looks like it would run after the metadetect one, so maybe put it in the same run as that?  You would just have to add the line to .github/workflows/ci.yml

**Review by empEvil — APPROVED** [review]
In that case I don't have any issues with us merging this.

**Comment by elvinpoole** [comment]
ok great, I'll look into adding to the CI now

**Comment by elvinpoole** [comment]
is the shifter on nersc updated to use the healpix module @joezuntz ?

**Comment by joezuntz** [comment]
I've just started pulling now, it should take < 5 min.

**Comment by joezuntz** [comment]
@elvinpoole  looks good - happy for me to merge?

**Pull request merged**